init.lua

```diff
@@ -0,0 +1,572 @@
+map("n", "<Leader>x", function()
+  if vim.o.filetype == "vim" or vim.o.filetype == "lua" then
+    vim.cmd("write | source %")
+  end
+end, "Re-executes the current vim / lua file.")
+
+---}}}
+
+---{{{ Plugins
+
+---{{{ Plugin manager
+
+-- This makes sure balls.nvim (your plugin manager) is installed.
+--
+-- See `:help balls.nvim`.
+local balls_path = vim.fs.joinpath(vim.fn.stdpath("config"), "/pack/balls/start/balls.nvim")
+
+if not vim.uv.fs_stat(balls_path) then
+  local command = { "git", "clone", "https://github.com/AlphaKeks/balls.nvim.git", balls_path }
+  local opts = { text = true }
+  local result = vim.system(command, opts):wait()
+
+  if result.code ~= 0 then
+    error("Failed to install balls.nvim! " .. vim.inspect(result))
+  end
+
+  vim.cmd.packloadall()
+  vim.notify("Installed balls.nvim! Run `:BallsInstall` to install your plugins!")
+end
+
+local ok, balls = pcall(require, "balls")
+
+if not ok then
+  vim.notify("balls.nvim not found. Plugins are disabled!")
+  return
+end
+
+---}}}
+
+---{{{ Colorscheme
+
+balls.register({
+  url = "https://github.com/catppuccin/nvim.git",
+  name = "catppuccin",
+})
+
+local catppuccin_installed, catppuccin = pcall(require, "catppuccin")
+
+if catppuccin_installed then
+  -- See `:help catppuccin-configuration` if you want to do any more configuration.
+  catppuccin.setup({
+    flavour = "mocha",
+  })
+
+  -- neovim also comes with a bunch of builtin colorschemes. Run `:colorscheme ` and hit tab!
+  -- Most of them are pretty hideous but `habamax` and `quiet` are pretty good.
+  --
+  -- Catppuccin is better though :)
+  vim.cmd.colorscheme("catppuccin")
+end
+
+--}}}
+
+---{{{ Treesitter
+---
+--- Treesitter is responsible for better syntax highlighting, indentation, folding, and more!
+--- The way it is setup it will install any missing parsers for the current file automatically.
+--- If you want to explicitly install a parser run `:TSInstall <language>` or put it in the
+--- `ensure_installed` table below.
+---
+--- Treesitter requires a C compiler to compile the parsers. If you're on Windows you might not have
+--- a C compiler installed, or the one you have doesn't work very well with treesitter. From
+--- experience zig has proven to be the most reliable on Windows, so if you have issues with
+--- treesitter I recommend you install zig:
+---
+---   <https://ziglang.org/download/>
+---
+--- And read this section of the nvim-treesitter docs:
+---
+---   <https://github.com/nvim-treesitter/nvim-treesitter/wiki/Windows-support#which-c-compiler-will-be-used>
+---
+--- And tell treesitter to use zig for compiling parsers.
+---
+--- To make sure everything goes smoothly I also recommend deleting old failed parsers before trying
+--- to recompile them with zig. To do that, simply delete the following directory starting at the
+--- root of your nvim config: `pack/balls/start/nvim-treesitter/parser`
+---
+--- Then try to compile the parsers again.
```

Comment:
They shouldn't have any

Reply:
If treesitter tries to compile parsers with whatever compiler they already have, they might be left with broken parsers.

Reply:
I thought people would read the file before using it but if that's not the case then yeah fair